### PR TITLE
add-http-exporter + UTs

### PR DESCRIPTION
**Pull request by Bezbran**
## **Type**
enhancement, tests

___

## **Description**
- Implemented an `HTTPExporter` to send security alerts over HTTP.
  - Configurable via `HTTPExporterConfig` with URL, headers, timeout, method, and rate limit.
  - Supports sending rule-based alerts and malware detection alerts.
  - Includes rate limiting to control the number of alerts sent per minute.
- Added comprehensive unit tests for the new `HTTPExporter`.
  - Tests cover sending rule alerts, rate limiting behavior, and malware alert sending.
  - Configuration validation tests ensure proper setup of the exporter.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>http_exporter.go</strong><dd><code>Implement HTTP Exporter for Alerts</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/exporters/http_exporter.go
<li>Introduced <code>HTTPExporter</code> for sending alerts over HTTP.<br> <li> Added <code>HTTPExporterConfig</code> to configure the HTTP exporter.<br> <li> Implemented methods for sending rule and malware alerts.<br> <li> Added rate limiting for alerts sent per minute.<br>

</details>
    

  </td>
  <td><a href="https:/armosec/kubecop/pull/186/files#diff-6e04fd42d767812ea2855370a21a524371930b320544c0ee0954e1500242fbbd">+262/-0</a>&nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Tests
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>http_exporter_test.go</strong><dd><code>Unit Tests for HTTP Exporter</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/exporters/http_exporter_test.go
<li>Added unit tests for <code>HTTPExporter</code> functionalities.<br> <li> Tested sending rule alerts, handling rate limits, and sending malware <br>alerts.<br> <li> Validated HTTP exporter configuration.<br>

</details>
    

  </td>
  <td><a href="https:/armosec/kubecop/pull/186/files#diff-89585231dfa94efdc4cac61ef6426f59bf494acd49c810bb29d56e16574fd350">+231/-0</a>&nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: chart/kubecop/values.yaml [hunk 1]
```diff
@@ -30,6 +30,9 @@ kubecop:
   alertmanager:
     enabled: false
     endpoints: "localhost:9093"
+  httpEndpoint:
+    enabled: false
+    url: "http://synchronizer.kubescape.svc.cluster.local/apis/v1/kubescape.io/v1/RuntimeAlerts"
```
Comment (amitschendel) [inline chart/kubecop/values.yaml:35]: Why it is http? we don't encrypt the traffic between our micro services?

File: pkg/exporters/http_exporter.go [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+		PodNamespace:  failedRule.Event().Namespace,
+		PodName:       failedRule.Event().PodName,
+		HostName:      exporter.Host,
+		NodeName:      exporter.NodeName,
+		RuleAlert: RuleAlert{
+			Severity:       failedRule.Priority(),
+			FixSuggestions: failedRule.FixSuggestion(),
+			PID:            failedRule.Event().Pid,
+			PPID:           failedRule.Event().Ppid,
+			CommandName:    failedRule.Event().Comm,
```
Comment (amitschendel) [inline pkg/exporters/http_exporter.go:163]: ProcessName

**Review by amitschendel — CHANGES_REQUESTED** [review]

File: chart/kubecop/values.yaml (hunk 1 above)
Comment (Bezbran) [reply chart/kubecop/values.yaml:35]: no we don't

Comment (amitschendel) [reply chart/kubecop/values.yaml:35]: Very bad, Very bad, Very bad, Very Very Very Very Very Very bad.

**Review by amitschendel — APPROVED** [review]

**Pull request merged**